packages/ckeditor5-ui/src/focuscycler.ts

```diff
@@ -274,6 +276,97 @@ export default class FocusCycler extends /* #__PURE__ */ EmitterMixin() {
 		}
 	}
 
+	/**
+	 * Allows for creating continuous focus cycling across multiple focus cyclers and their collections of {@link #focusables}.
+	 *
+	 * It starts listening to the {@link module:ui/focuscycler~FocusCyclerForwardCycleEvent} and
+	 * {@link module:ui/focuscycler~FocusCyclerBackwardCycleEvent} events of the chained focus cycler and engages,
+	 * whenever the user reaches the last (forwards navigation) or first (backwards navigation) focusable view
+	 * and would normally start over. Instead, the navigation continues on the higher level (flattens).
+	 *
+	 * For instance, for the following nested focus navigation structure, the focus would get stuck the moment
+	 * the AB gets focused and its focus cycler starts managing it:
+	 *
+	 *	   ┌────────────┐   ┌──────────────────────────────────┐   ┌────────────┐
+	 *	   │ AA         │   │ AB                               │   │ AC         │
+	 *	   │            │   │                                  │   │            │
+	 *	   │            │   │    ┌─────┐  ┌─────┐  ┌─────┐     │   │            │
+	 *	   │            │   │ ┌──► ABA ├──► ABB ├──► ABC ├───┐ │   │            │
+	 *	┌──►            ├───► │  └─────┘  └─────┘  └─────┘   │ ├───►            ├──┐
+	 *	│  │            │   │ │                              │ │   │            │  │
+	 *	│  │            │   │ │                              │ │   │            │  │
+	 *	│  │            │   │ └──────────────────────────────┘ │   │            │  │
+	 *	│  │            │   │                                  │   │            │  │
+	 *	│  └────────────┘   └──────────────────────────────────┘   └────────────┘  │
+	 *	│                                                                          │
+	 *	└──────────────────────────────────────────────────────────────────────────┘
```

Comment:
```suggestion
	 *
	 *	   ┌────────────┐   ┌──────────────────────────────────┐   ┌────────────┐
	 *	   │ AA         │   │ AB                               │   │ AC         │
	 *	   │            │   │                                  │   │            │
	 *	   │            │   │    ┌─────┐  ┌─────┐  ┌─────┐     │   │            │
	 *	   │            │   │ ┌──► ABA ├──► ABB ├──► ABC ├───┐ │   │            │
	 *	   │            ├───► │  └─────┘  └─────┘  └─────┘   │ │   │            │
	 *	   │            │   │ │                              │ │   │            │  
	 *	   │            │   │ │                              │ │   │            │  
	 *	   │            │   │ └──────────────────────────────┘ │   │            │  
	 *	   │            │   │                                  │   │            │  
	 *	   └────────────┘   └──────────────────────────────────┘   └────────────┘  
	 *	                                                                           
```

Reply:
My suggestion is to change this diagram so it is more clear what happened here: focus got from AA, to AB, then to its children and never got to AC. Maybe it needs some slight clarification in the docs description too.

Reply:
Otherwise, really nice docs for a non-trivial mechanism.